### PR TITLE
Feature/add annotations for prometheus

### LINKED ISSUE #48: Expose metrics during tests execution
We already deployed Prometheus in Dist-Tests cluster and tested how we can get Codex metrics.
<details>
<summary>screenshots</summary>

Screenshot 2023-08-22 at 16 41 15

Screenshot 2023-08-22 at 17 01 05
</details>

**Configuration**
1. Configure Codex to expose the metrics
2. Expose Codex container metrics port to be accessible for scraping
3. Add Codex Pod annotations to use Prometheus Service Discovery for metrics scraping

**Manifest**
```yaml
---
apiVersion: v1
kind: Pod
metadata:
  name: codex-metrics
  namespace: default
  annotations:
    prometheus.io/scrape: "true"
  labels:
    app: codex
    tests-type: dist-tests
spec:
  containers:
  - name: codex
    image: codexstorage/nim-codex:latest-dist-tests
    imagePullPolicy: Always
    env:
    - name: CODEX_METRICS
      value: "true"
    - name: CODEX_METRICS_ADDRESS
      value: "0.0.0.0"
    - name: CODEX_METRICS_PORT
      value: "8008"
    ports:
      - name: metrics
        containerPort: 8008
        protocol: TCP

```

### PR DESCRIPTION
**Pull request by veaceslavdoina**
This PR adds Prometheus annotations, required for metrics scraping, to the Codex Pods when metrics are enabled.

Also, it was decided to refactor how do we handle Pods Labels and Annotations assignment.

Closes #48.

### CODE REVIEW

File: CodexNetDeployer/Deployer.cs [hunk 1]
```diff
@@ -45,19 +45,16 @@ public CodexDeployment Deploy()
...
-                if (result != null) startResults.Add(result);
+                var container = codexStarter.Start(i);
+                if (container != null) codexContainers.Add(container);
             }
 
-            var (prometheusContainer, grafanaStartInfo) = StartMetricsService(lifecycle, setup, startResults.Select(r => r.Container));
+            var (prometheusContainer, grafanaStartInfo) = StartMetricsService(lifecycle, setup, codexContainers);
 
-            CheckPeerConnectivity(startResults);
-            CheckContainerRestarts(startResults);
```
Comment (benbierens) [inline CodexNetDeployer/Deployer.cs:58]: There seem to be quite a few changes here that are not related to the annotations or pod labels. It looks like a merge/conflict resolution didn't go so well. Please check.

**Comment by veaceslavdoina** [comment]
I did reset my latest merge
```shell
git reset --hard HEAD~1
git push origin fix-feature/add-annotations-for-prometheus:feature/add-annotations-for-prometheus --force
```

Now, GitHub shows me just a single conflict
Screenshot 2023-09-01 at 17 05 39
